tests/conftest.py

```diff
@@ -292,6 +292,12 @@ def run_and_assert_assert_pytest_result(
         # checked by the overloads
         result = self.run_pytest(*pytest_args or [], subprocess=subprocess, plugins=plugins)  # pyright:ignore[reportArgumentType]
 
+        if not errors:
+            for line in result.errlines:
+                if line.startswith("[ ERROR ] "):
+                    raise Exception(
+                        f"robot error detected in a test that expected no errors: {line}"
+                    )
```

Comment:
```suggestion
        if not errors and any(line.startswith("[ ERROR ] ") for line in result.errlines):
                    raise Exception(
                        f"robot error detected in a test that expected no errors: {line}"
                    )
```

Reply:
need the `line` variable in the exception tho